src/Rootfs.jl

```diff
@@ -350,19 +412,19 @@ function choose_shards(p::Platform;
             compilers::Vector{Symbol} = [:c],
             rootfs_build::VersionNumber=v"2020.01.07",
             ps_build::VersionNumber=v"2020.01.15",
-            GCC_builds::Vector{VersionNumber}=available_gcc_builds,
-            LLVM_builds::Vector{VersionNumber}=available_llvm_builds,
+            GCC_builds::Vector{GCCBuild}=available_gcc_builds,
+            LLVM_builds::Vector{LLVMBuild}=available_llvm_builds,
             Rust_build::VersionNumber=v"1.18.3",
             Go_build::VersionNumber=v"1.13",
             archive_type::Symbol = (use_squashfs ? :squashfs : :unpacked),
             bootstrap_list::Vector{Symbol} = bootstrap_list,
             # We prefer the oldest GCC version by default
-            preferred_gcc_version::VersionNumber = GCC_builds[1],
-            preferred_llvm_version::VersionNumber = LLVM_builds[end],
+            preferred_gcc_version::VersionNumber = getversion(GCC_builds[1]),
+            preferred_llvm_version::VersionNumber = getversion(LLVM_builds[end]),
```

Comment:
Let's add a comment here saying "because GCC has lots of compatibility issues, we always default to the earliest version possible.  Because LLVM doesn't have compatibilty issues, we always default to the newest version possible".

Reply:
Awesome, I wondered why the we had the different choice.